src/AzureClient/AzureClient.cs

```diff
@@ -210,19 +229,58 @@ public async Task<ExecutionResult> GetConnectionStatusAsync(IChannel channel)
             IChannel channel)
         {
-            // TODO: This should also print the list of available targets to the IChannel.
-            return ActiveTargetName.ToExecutionResult();
+            if (AvailableProviders == null)
+            {
+                channel.Stderr("Please call %azure.connect before getting the execution target.");
+                return AzureClientError.NotConnected.ToExecutionResult();
+            }
+
+            if (ActiveTarget == null)
+            {
+                channel.Stderr("No execution target has been specified. To specify one, run:\n%azure.target <target name>");
+                channel.Stdout($"Available execution targets: {ValidExecutionTargetsDisplayText}");
+                return AzureClientError.NoTarget.ToExecutionResult();
+            }
+
+            channel.Stdout($"Current execution target: {ActiveTarget.TargetName}");
+            channel.Stdout($"Available execution targets: {ValidExecutionTargetsDisplayText}");
+            return ActiveTarget.TargetName.ToExecutionResult();
         }
 
         /// <inheritdoc/>
         public async Task<ExecutionResult> SetActiveTargetAsync(
             IChannel channel,
+            IReferences references,
             string targetName)
         {
-            // TODO: Validate that this target name is valid in the workspace.
-            // TODO: Load the associated provider package.
-            ActiveTargetName = targetName;
-            return $"Active target is now {ActiveTargetName}".ToExecutionResult();
+            if (AvailableProviders == null)
+            {
+                channel.Stderr("Please call %azure.connect before setting an execution target.");
+                return AzureClientError.NotConnected.ToExecutionResult();
+            }
+
+            // Validate that this target name is valid in the workspace.
+            if (!AvailableTargets.Any(target => targetName == target.Id))
+            {
+                channel.Stderr($"Target name {targetName} is not available in the current Azure Quantum workspace.");
+                channel.Stdout($"Available execution targets: {ValidExecutionTargetsDisplayText}");
+                return AzureClientError.InvalidTarget.ToExecutionResult();
+            }
+
+            // Validate that we know which package to load for this target name.
+            var executionTarget = AzureExecutionTarget.Create(targetName);
+            if (executionTarget == null)
+            {
+                channel.Stderr($"Target name {targetName} does not support executing Q# jobs.");
+                channel.Stdout($"Available execution targets: {ValidExecutionTargetsDisplayText}");
+                return AzureClientError.InvalidTarget.ToExecutionResult();
+            }
+
+            // Set the active target and load the package.
+            ActiveTarget = executionTarget;
+            await references.AddPackage(ActiveTarget.PackageName);
```

Comment:
Normally the `%package` magic prints out a lot of status information as it goes, since package loading can take a while on slow connections or when the NuGet cache needs to be populated for the first time. Would it be worth making transient displays here to help communicate to users that a package is being loaded?

Reply:
Yes, that's a good suggestion. Ideally I'd like to just call the `%package` magic directly here, but both it and the `IMagicSymbolResolver` are in the `Kernel` project and I can't access them directly.

I'll do the bare minimum for now and add a line of output indicating that the package and its dependencies are being loaded.

Reply:
Makes sense. Perhaps it would be good to put a task on the backlog to refactor and consolidate the two package load calls, then?

Reply:
Sounds good! I've added a task.